request_adapter.go

```diff
@@ -19,19 +19,19 @@ type ErrorMappings map[string]s.ParsableFactory
 // RequestAdapter is the service responsible for translating abstract RequestInformation into native HTTP requests.
 type RequestAdapter interface {
 	// SendAsync executes the HTTP request specified by the given RequestInformation and returns the deserialized response model.
-	SendAsync(context context.Context, requestInfo *RequestInformation, constructor s.ParsableFactory, responseHandler ResponseHandler, errorMappings ErrorMappings) (s.Parsable, error)
+	SendAsync(context *context.Context, requestInfo *RequestInformation, constructor s.ParsableFactory, responseHandler ResponseHandler, errorMappings ErrorMappings) (s.Parsable, error)
```

Comment:
context is already an interface, and interfaces are already pointer based. no need to add a pointer symbol here https://pkg.go.dev/context#Context